_posts/2021-11-19-connecting-java-high-level-rest-client-with-opensearch-over-https.markdown

```diff
@@ -0,0 +1,152 @@
+---
+layout: post
+title:  "Connecting Java High-level REST client with OpenSearch over HTTPS"
+authors: 
+  - setiah
+date: 2021-11-19 01:01:01 -0700
+categories: 
+  - technical-post
+twittercard:
+  description: "This post walks through the steps to configure Java High-level REST client to connect with OpenSearch over HTTPS"
+---
+
+If you have ever used OpenSearch with a java application, you might have come across OpenSearch Java high-level REST client. Java high-level REST client provides OpenSearch APIs as methods, and makes it easier for a java application to interact with OpenSearch using request/response objects.
+
+In this blog, we will see how to configure the Java high-level REST client (a.k.a JHLRC) to connect with an OpenSearch cluster securely. We will set up a local OpenSearch server using the default distribution which comes with security plugin pre-installed, and use JHLRC client in a sample Java application to interact with the server over HTTPS. I have used an Ubuntu 18.04 machine, but it should work for other linux distributions as well.
```

Comment:
What about Windows distributions!? :)

Reply:
Tbh I'm not sure if we have a proper distribution available for Windows. The docker images for OpenSearch all seem to be [linux based (AL2/Centos)](https://github.com/opensearch-project/opensearch-build/tree/56f4092bdac8ee676ac9610a594d6657149e06be/docker/ci/dockerfiles), which I suspect may not be the right ones to run on windows (?)

Reply:
We don't. I do think we have users that want to write Java code that runs on Windows, so the post should probably do a clean(er) job of separating what's setup of a cluster vs. the actual purpose of the post, aka getting the client to work.

Reply:
re-worded it. Hope it is more clear now.